src/flatpickr.js

```diff
@@ -1801,8 +1801,9 @@ function Flatpickr(element, config) {
 				.map(dObj => self.formatDate(dObj, self.config.altFormat))
 				.join(joinChar);
 		}
-
-		triggerEvent("ValueUpdate");
+		if (triggerChange !== false) {
```

Comment:
@zerkalica yes, but there's `onChange` with `triggerChangeEvent` flag specifically for those scenarios.

Its reasonable to expect that calling `setDate` would trigger `onValueUpdate`, and I hope even dummy programmers are familiar with the concept of recursion.

Apart from that, the rest of the PR looks good.